src/libraries/System.Private.CoreLib/src/System/Environment.OSVersion.MacCatalyst.cs

```diff
@@ -7,7 +7,16 @@ public static partial class Environment
     {
         private static OperatingSystem GetOSVersion()
         {
-            Version version = new Version(Interop.Sys.iOSSupportVersion());
+            Version version = Version.Parse(Interop.Sys.iOSSupportVersion());
+
+            // Check if build and revision are -1 and default them to 0
+            int major = version.Major;
+            int minor = version.Minor;
+            int build = version.Build >= 0 ? version.Build : 0;
+            int revision = version.Revision >= 0 ? version.Revision : 0;
+
+            version = new Version(major, minor, build, revision);
```

Comment:
we don't need to create a new object, we can just set version.Build and version.Revision.

Reply:
The properties are `readonly` and I think cannot be assigned.